src/uu/ls/src/ls.rs

```diff
@@ -1253,63 +1301,85 @@ impl PathData {
             }
             Dereference::None => false,
         };
+
         let ft = match file_type {
             Some(ft) => OnceCell::from(ft.ok()),
             None => OnceCell::new(),
         };
 
         let security_context = if config.context {
-            get_security_context(config, &p_buf, must_dereference)
+            get_security_context(config, &p_buf, &must_dereference)
         } else {
             String::new()
         };
 
+        #[allow(clippy::match_like_matches_macro)]
+        let store_md: bool = match config.sort {
+            Sort::None => false,
+            _ => true,
+        };
+
         Self {
             md: OnceCell::new(),
             ft,
             display_name,
             p_buf,
             must_dereference,
             security_context,
+            store_md,
         }
     }
 
-    fn md(&self) -> Option<&Metadata> {
-        self.md
-            .get_or_init(|| get_metadata(&self.p_buf, self.must_dereference).ok())
-            .as_ref()
+    #[allow(clippy::redundant_clone)]
+    fn md(&self) -> Option<Metadata> {
+        let res = if self.store_md {
+            self.md
+                .get_or_init(|| {
+                    get_metadata(&self.p_buf.clone(), &self.must_dereference.clone()).ok()
```

Comment:
Why do you do all this complicated stuff with a `bool`? It's a `Copy` type, so you essentially never have to borrow or clone it at all.